library/std/src/sys/windows/process.rs

```diff
@@ -31,32 +32,72 @@ use libc::{c_void, EXIT_FAILURE, EXIT_SUCCESS};
 // Command
 ////////////////////////////////////////////////////////////////////////////////
 
-#[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd)]
+#[derive(Clone, Debug, Eq)]
 #[doc(hidden)]
-pub struct EnvKey(OsString);
+pub struct EnvKey {
+    os_string: OsString,
+    utf16: Vec<u16>,
```

Comment:
It is unfortunate that `OsString` on Windows is backed by an WTF8 encoded `Vec<u8>`, instead of an `Vec<u16>`. Normally in Windows API calls the osstring is only converted to an u16 buffer right before the call (see almost every function in `sys::windows::fs`). However converting and having to allocate for every `==` would be expensive, so storing the utf16 encoding seems like a reasonable solution. Could you add a comment to the field explaining that this is a workaround?

Reply:
(@CDirkx Unrelated to this PR: Note that this happens even on Unix, because OsString is not zero terminated. So every time one has to be passed to the OS, it gets copied to a new allocation to add a zero terminator, which is deallocated again right afterwards. Unfortunately, 'OsString' means 'can represent every string that the OS accepts', not 'a string that the OS can directly use'. We can't change this unless we start a 'std 2.0' or something.)